### ISSUE TITLE
Keycloak 11 removes token_introspection_endpoint from discovery endpoint

### ISSUE THREAD

**Issue by jhkuperus** [eclipse-vertx/vertx-auth#434]
### Version

3.9.4

### Context

Our operations team recently upgraded Keycloak to version 11. This caused all of our Vertx-services to suddenly fail with `403` responses with `Invalid path` as the underlying message. After some analysis, the `Invalid path` message was a response from Keycloak, which Vertx-auth was trying to connect at `introspectionPath == null`.

It appears Keycloak 11 has removed the deprecated field `token_introspection_endpoint` in favor of the field `introspection_endpoint` (based on RFC-8414). This causes `OpenIDConnectAuth` to read a `null` value for the token introspection endpoint and later break.

### Do you have a reproducer?

I'll curate a discovery-document for this issue.

### Steps to reproduce

1. Deploy a Keycloak 11 and use discovery for configuration
2. 
3. ...
4. ...

### Extra

* OS : Ubuntu 20.04
* JVM : 

```
openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment (build 11.0.7+10-post-Ubuntu-3ubuntu1)
OpenJDK 64-Bit Server VM (build 11.0.7+10-post-Ubuntu-3ubuntu1, mixed mode, sharing)
```

**Comment by jhkuperus** [comment]
Apologies, I pressed enter too soon. Still writing the bug-report properly.

**Comment by pmlopes** [comment]
@jhkuperus could you attach an example discovery file with the new keys. It would be great to write a regression test.

**Comment by jhkuperus** [comment]
Here's our the respsonse from our Keycloak on `/auth/realms/test-realm/.well-known/openid-configuration`:

[keycloak-11-discovery.json.zip](https://github.com/vert-x3/vertx-auth/files/5516290/keycloak-11-discovery.json.zip)

(The actual realm name and hostname have been replaced)

**Comment by jhkuperus** [comment]
By the way, I'm trying to get the `OAuth2KeycloakIT` to run, but it just keeps failing on `OAuth2KeycloakIT.lambda$setUp$0:54 Expected true`, even with the Keycloak 6 docker image that you provide in the fixtures. Any hints?

**Comment by pmlopes** [comment]
It works for me.

```
docker build -t vertx-test-keycloak vertx-auth-oauth2/src/test/fixtures
docker run -d -p 8888:8080 -p 9443:8443 vertx-test-keycloak
```
from the root of the project. Upgrading to 11 would be interesting but it also requires a new dump of the fixtures which aren't comatible with newer versions.

**Comment by pmlopes** [comment]
I've added a test to confirm that introspection works with keycloak. Aparently keycloak 6 supported both keys already on 6. Switiching to the official one seems to be the better choice instead of relying on some legacy config.

**Comment by jhkuperus** [comment]
So, is this patched for a next 3.9.x-version? I prepared a version that read the RFC-8414 field first and tried to fallback on the legacy one if it read `null`. Then again, if most mainstream versions of keycloak already had the RFC-field as well, switching would indeed be the best solution.

I also looked over the 4.0 code couldn't find a reference to the legacy field there. So hopefully we won't run into this again when we migrate to 4.0 soonish. 

Thank you for your swift action!
